pkg/model/openstackmodel/firewall.go

```diff
@@ -539,8 +539,11 @@ func (b *FirewallModelBuilder) getExistingRules(sgMap map[string]*openstacktasks
 
 }
 
-func (b *FirewallModelBuilder) addDefaultEgress(c *fi.ModelBuilderContext, sgMap map[string]*openstacktasks.SecurityGroup) {
-	for _, sg := range sgMap {
+func (b *FirewallModelBuilder) addDefaultEgress(c *fi.ModelBuilderContext, sgMap map[string]*openstacktasks.SecurityGroup, useVIPACL bool) {
+	for name, sg := range sgMap {
+		if useVIPACL && name == b.Cluster.Spec.MasterPublicName {
```

Comment:
when using VIPACL the api.<clustername> securitygroup is not created at all, so we should not add these rules. See https://github.com/kubernetes/kops/blob/master/pkg/model/openstackmodel/firewall.go#L573-L584